spec.bs

```diff
@@ -2921,6 +2926,188 @@ This specification defines two [=policy-controlled features=] identified by the
+
+<h3 id=ad-auction-signals-header>`Ad-Auction-Signals` header</h3>
+
+The <dfn http-header><code>Ad-Auction-Signals</code></dfn> response header provides value
+of a JSON array of dictionaries, each with an `adSlot` key. Protected Audience non-component,
+component, and top-level auctions may specify which signals to load by the `adSlot` key.
+</div>
+
+<div algorithm="fetch auction signals response patch">
+The following step will be added to the [=http-network-or-cache fetch=] algorithm, before step
+"Return |response|.":
+
+1. If |response| is not null, then run [=update captured headers=] with [=this=]'s
+  [=relevant global object=]'s [=associated Document's=] [=node navigable's=]
+  [=traversable navigable's=] [=captured ad auction headers=], |response|'s
+  [=response/header list=], and <var ignore>request</var>'s [=request/URL=]'s [=url/origin=].
+
+</div>
+
+<div algorithm="fetch update captured headers patch">
+The following algorithm will be added to the [[FETCH#fetching]] section:
+
+<h3 id=update-captured-headers>Update captured headers</h3>
+
+  To <dfn id=concept-update-captured-headers>update captured headers</dfn> with a [=captured ad
+  auction headers=] |storedHeaders|, [=header list=] |responseHeaders|, and [=origin=]
+  |requestOrigin|:
+  1. Let |adAuctionSignals| be the result of [=header list/get=] with [:Ad-Auction-Signals:],
+    |responseHeaders|.
+  1. If |adAuctionSignals| is null, return.
+  1. [=header list/Delete=] "[:Ad-Auction-Signals:]" from |responseHeaders|.
+
+    NOTE: This step prevents the header value from being used outside the intended auctions --
+    that is, scripts making the {{WindowOrWorkerGlobalScope/fetch()}} request aren't able to load
+    the header value.
+  1. Let |parsedSignals| be the result of [=parsing JSON bytes to an Infra value=], given
+    |adAuctionSignals|.
+  1. If |parsedSignals| is failure, return.
+  1. If |parsedSignals| is not a [=list=], return.
+  1. [=list/For each=] |signal| in |parsedSignals|:
+    1. If |signal| is not an [=ordered map=], [=iteration/continue=].
+    1. If |signal|["`adSlot`"] doesn't exist, [=iteration/continue=].
+    1. Create a new [=direct from seller signals key=] |signalsKey|, with its
+      [=direct from seller signals key/seller=] set to |requestOrigin| and its
+      [=direct from seller signals key/ad slot=] set to |signal|["`adSlot`"].
+    1. Create a new [=direct from seller signals=] |processedSignals|.
+    1. [=map/Remove=] |signal|["`adSlot`"].
+    1. [=map/For each=] |key| → |value| of |signal|:
+      1. Switch on |key|:
+        <dl class=switch>
+          <dt>"`sellerSignals`"
+          <dd>
+          1. Set |processedSignals|'s [=direct from seller signals/seller signals=] to the result
+            of [=serializing an Infra value to a JSON string=], given |value|. If this
+            [=exception/throws=], [=iteration/continue=].

```

Comment:
Related: https://github.com/WICG/turtledove/pull/771#discussion_r1324519867. I don't think these can throw, right? This and it looks like two more instances below.